logic/resolvable/RetrievableTest.java

```diff
@@ -137,69 +137,82 @@ public void test_multiple_value_constraints_in_retrievable() {
         Set<Retrievable> retrievables = Retrievable.extractFrom(parse("{ $a has $b; $b > 5; $b < 10; }"), concludables);
         assertEquals(1, concludables.size());
         assertEquals(set(parse("{ $b > 5; $b < 10; }")),
-                     retrievables.stream().map(Retrievable::pattern).collect(Collectors.toSet()));
+                     iterate(retrievables).map(Retrievable::pattern).toSet());
     }
 
     @Test
-    public void test_disconnected_conjunction_in_retrievable() {
+    public void test_only_disconnected_retrievables_are_separated() {
         Set<Retrievable> retrievables = Retrievable.extractFrom(parse("{ $a = 7; $b > 5; $b < 10; }"), set());
         assertEquals(set(parse("{ $b > 5; $b < 10; }"),
                          parse("{ $a = 7; }")),
-                     retrievables.stream().map(Retrievable::pattern).collect(Collectors.toSet()));
+                     iterate(retrievables).map(Retrievable::pattern).toSet());
+    }
+
+    @Test
+    public void test_retrievables_split_by_concludables_are_separated() {
```

Comment:
Can we sanitise this example to be less random and more in-line with the other tests? Also the test just says it tests the splitting, but is this now the test to specifically catch that retrievables are split at the labelled type variables? If so let's add that to the name.

Reply:
done